### PR TITLE
drivers: can: drop POSIX from the native Linux SocketCAN driver name

### PR DESCRIPTION
**Pull request by henrikbrixandersen**
Rename the native Linux SocketCAN driver to reflect that it can can now be used in both native_posix and native_sim (with or without an embedded C-library).

### CODE REVIEW

**Review by kartben — DISMISSED** [review]

**Comment by jukkar** [comment]
I am just pondering here that should we even drop the `linux` from the name too. So calling things `native_sim` and perhaps even `nsim` in apis. WDYT

**Comment by henrikbrixandersen** [comment]
> I am just pondering here that should we even drop the `linux` from the name too. So calling things `native_sim` and perhaps even `nsim` in apis. WDYT

This particular driver is Linux-only, so I think we should reflect that in the name. Later, other "native" CAN drivers for other host platforms may be added.

**Review by martinjaeger — DISMISSED** [review]
I remember we had a longer discussion about the name in the original PR that introduced the driver. I never really liked the `native_posix_linux` and I'm very happy with `native_linux` only.

Thanks for implementing the change. Looks good.

File: drivers/can/can_native_linux.c [hunk 1]
```diff
@@ -383,46 +384,46 @@ static int can_npl_recover(const struct device *dev, k_timeout_t timeout)
 }
 #endif /* CONFIG_CAN_AUTO_BUS_OFF_RECOVERY */
 
-static void can_npl_set_state_change_callback(const struct device *dev,
-					      can_state_change_callback_t cb,
-					      void *user_data)
+static void can_native_linux_set_state_change_callback(const struct device *dev,
```
Comment (aescolar) [inline drivers/can/can_native_linux.c:387]: Just bike-shedding: Quite long names for static functions, good that we don't have the 80 chars limit anymore :)
(just ignore unless you really feel like changing it)

**Comment by aescolar** [comment]
> I am just pondering here that should we even drop the linux from the name too. So calling things native_sim and perhaps even nsim in apis. WDYT

I would rather avoid having the board name (native_sim) in the driver unless the driver is really coupled with the board itself. That proved not a good choice for the native_posix ones.

**Review by aescolar — DISMISSED** [review]
Thanks :)

**Comment by fabiobaltieri** [comment]
@henrikbrixandersen this needs a rebase

**Comment by henrikbrixandersen** [comment]
Rebased to fix a merge conflict in the migration guide.

**Review by fabiobaltieri — APPROVED** [review]

**Review by aescolar — APPROVED** [review]

**Review by martinjaeger — APPROVED** [review]

**Review by kartben — APPROVED** [review]

**Pull request merged**